algorithm.go

```diff
@@ -72,6 +75,44 @@ func (a Algorithm) String() string {
 	}
 }
 
+func (a Algorithm) IsSupported() bool {
+	return (a >= -39 && a <= -35) || (a >= -8 && a <= -7)
+}
+
+// MarshalCBOR marshals the Algorithm as a CBOR int.
+func (a Algorithm) MarshalCBOR() ([]byte, error) {
+	return encMode.Marshal(int64(a))
+}
+
+// UnmarshalCBOR populates the Algorithm from the provided CBOR value (must be
+// int or tstr).
+func (a *Algorithm) UnmarshalCBOR(data []byte) error {
+	var raw intOrStr
+
+	if err := raw.UnmarshalCBOR(data); err != nil {
+		return fmt.Errorf("invalid algorithm value: %w", err)
+	}
+
+	if raw.IsString() {
+		v := algorithmFromString(raw.String())
+		if v == AlgorithmInvalid {
+			return fmt.Errorf("unknown algorithm value %q", raw.String())
+		}
+
+		*a = v
+	} else {
+		v := raw.Int()
+		*a = Algorithm(v)
+
+		if !a.IsSupported() {
+			return fmt.Errorf("unknown algorithm value %d", v)
+		}
```

Comment:
Rejecting _unsupported_ algorithm makes `go-cose` losing its crypto agility to support other algorithms. Anyone can extend `go-cose` to support those _unsupported_ algorithms by implementing corresponding [Signer](https://pkg.go.dev/github.com/veraison/go-cose#Signer)s.

For example, `RS256` (-257) is allowed by COSE although it is not recommended (see [RFC8812](https://www.rfc-editor.org/rfc/rfc8812.html#name-rsassa-pkcs1-v1_5-signature)).